lib/client.js

```diff
@@ -360,33 +359,25 @@ export default class Client {
   }
 
   updateIgnore() {
-    if (!this.ignoreFile) {
-      this.ignorePath = this.getFilePath(this.ignoreBaseName);
-      this.ignoreFile = new File(this.ignorePath);
-    }
-
-    if (!this.ignoreFile.existsSync()) {
-      this.ignoreFilter = false;
+    const ignorePath = this.getFilePath(this.ignoreBaseName);
+    const ignoreFile = new File(ignorePath);
+    if (!ignoreFile.existsSync()) {
       return false;
     }
-
-    if (this.ignoreFile.getBaseName() === this.ignoreBaseName) {
-      this.ignoreFilter = ignore().add(this.ignoreFile.readSync(true));
-      return true;
-    }
-
-    return false;
+    this.ignoreFilter = ignore().add(ignoreFile.readSync(true));
+    return true;
   }
 
-  checkIgnore(local) {
-    let haseIgnore = true;
+  checkIgnore(filepath) {
+    filepath = this.toRelative(filepath);
```

Comment:
Please no redefine the parameter.

Reply:
fix it!

Reply:
I found another mistake. I'll send the repair right away.

(If you start with "/", subfolder filtering will not work)